pkg/services/ngalert/notifier/file_store.go

```diff
@@ -85,10 +85,20 @@ func (fs *FileStore) Persist(ctx context.Context, filename string, st State) (in
 	return int64(len(bytes)), err
 }
 
+// Delete will remove the file from the filestore.
+func (fs *FileStore) Delete(ctx context.Context, filename string) error {
+	if fs.IsExists(filename) {
+		if err := os.Remove(fs.pathFor(filename)); err != nil {
+			return err
+		}
+	}
+	return fs.kv.Del(ctx, filename)
```

Comment:
I wonder if we should ignore the first error and keep proceed to db. The file can be locked by the process

Reply:
Perhaps we should inverse it, first delete from db (and return if there is an error), then from fs?